cpp/src/arrow/compute/kernels/test_util_internal.h

```diff
@@ -50,6 +51,15 @@ namespace compute {
 
 using DatumVector = std::vector<Datum>;
 
+class ComputeKernelEnvironment : public ::testing::Environment {
```

Comment:
I have another idea about taking use of this environment. I will send out a PR based on yours.

Reply:
Would you like to take a look? Thanks. @raulcd 

https://github.com/raulcd/arrow/pull/92